runtime/src/provisioner_factory.rs

```diff
@@ -0,0 +1,156 @@
+use std::{collections::BTreeMap, path::PathBuf};
+
+use async_trait::async_trait;
+use shuttle_common::{database, storage_manager::StorageManager, DatabaseReadyInfo};
+use shuttle_proto::provisioner::{
+    database_request::DbType, provisioner_client::ProvisionerClient, DatabaseRequest,
+};
+use shuttle_service::{Factory, ServiceName};
+use tonic::{transport::Channel, Request};
+use tracing::{debug, info, trace};
+use uuid::Uuid;
+
+/// Trait to make it easy to get a factory (service locator) for each service being started
+pub trait AbstractFactory<S: StorageManager>: Send + 'static {
```

Comment:
I don't think the abstract factory is still needed (it's no longer being send across threads). We can just create the factory below directly